it/src/test/scala/com/wavesplatform/it/sync/SmartContractsAtomicSwapSuite.scala

```diff
@@ -0,0 +1,237 @@
+import com.wavesplatform.it.api.SyncHttpApi._
+import com.wavesplatform.it.transactions.BaseTransactionSuite
+import com.wavesplatform.it.util._
+import com.wavesplatform.lang.v1.{Parser, TypeChecker}
+import com.wavesplatform.state._
+import com.wavesplatform.utils.dummyTypeCheckerContext
+import org.scalatest.CancelAfterFailure
+import play.api.libs.json.JsNumber
+import scorex.account.PrivateKeyAccount
+import scorex.transaction.{Proofs}
+import scorex.transaction.smart.SetScriptTransaction
+import scorex.transaction.smart.script.v1.ScriptV1
+import scorex.transaction.transfer._
+import scorex.crypto.encode.{Base58 => ScorexBase58}
+
+class SmartContractsAtomicSwapSuite extends BaseTransactionSuite with CancelAfterFailure {
+
+  private val BobBC1: String   = sender.createAddress()
+  private val AliceBC1: String = sender.createAddress()
+  private val swapBC1: String  = sender.createAddress()
+
+  private val BobBC2: String   = sender.createAddress()
+  private val AliceBC2: String = sender.createAddress()
+  private val swapBC2: String  = sender.createAddress()
+
+  private val transferAmount: Long = 1.waves
+  private val fee: Long            = 0.001.waves
+
+  private val secretText = "some secret message from Alice"
+  private val secretHash = ScorexBase58.encode(secretText.getBytes)
+
+  private val sc1 = {
+    val untyped = Parser(s"""
+    true
+      """.stripMargin).get.value
+    TypeChecker(dummyTypeCheckerContext, untyped).explicitGet()
+  }
+
+  private val sc2 = {
+    val untyped = Parser(s"""
+    let Alice = extract(addressFromString("${AliceBC2}")).bytes
+    let backToBob = extract(addressFromString("${BobBC2}")).bytes
+    let txRecipient = addressFromRecipient(tx.recipient).bytes
+
+    let txTransfer = if((txRecipient == Alice) && (sha256(tx.proof0) == base58'BN6RTYGWcwektQfSFzH8raYo9awaLgQ7pLyWLQY4S4F5') && (100 >= height)) then true else false
+    let backToBobAfterHeight = if ((height >= 101) && (txRecipient == txToBob)) then true else false
+
+    txTransfer || backToBobAfterHeight
+      """.stripMargin).get.value
+    TypeChecker(dummyTypeCheckerContext, untyped).explicitGet()
+  }
+
+  test("step0: balances initialization") {
+    val toAliceBC1TxId = sender.transfer(sender.address, AliceBC1, 10 * transferAmount, fee).id
+    nodes.waitForHeightAriseAndTxPresent(toAliceBC1TxId)
+
+    val toSwapBC1TxId = sender.transfer(sender.address, swapBC1, transferAmount, fee).id
+    nodes.waitForHeightAriseAndTxPresent(toSwapBC1TxId)
+
+    val toBobBC2TxId = sender.transfer(sender.address, BobBC2, 10 * transferAmount, fee).id
+    nodes.waitForHeightAriseAndTxPresent(toBobBC2TxId)
+
+    val toSwapBC2TxId = sender.transfer(sender.address, swapBC2, transferAmount, fee).id
+    nodes.waitForHeightAriseAndTxPresent(toSwapBC2TxId)
+  }
+
+  test("step1: create and setup smart contract SC1") {
+    val pkSwapBC1 = PrivateKeyAccount.fromSeed(sender.seed(swapBC1)).right.get
+    val script    = ScriptV1(sc1).explicitGet()
```

Comment:
Soo `pkSwapBC1` becomes open to everyone? And then you put money there? :)

Reply:
Nope, transfer to swap address at the next step